### PR TITLE
MK II. SWAT Suit

### PR DESCRIPTION
**Pull request by Steelpoint**
This PR sees the old mk2 SWAT suit sprites get some use. This PR creates official baseline MKII Swat Suit.

In 'lore' the MKII SWAT Suit is the baseline of which the Captains SWAT suit is split off from, and is intended as a 'frontline combat armour' for future soldiers. The older MKI SWAT suit is represented as an aged surplus item that cargo can order (and is the signature cargonian uniform) whereas the MKIII SWAT Suit is the hyperadvanced/prototype/expensive newer variant used by the secretive NT Deathsquad. 

screenshot 2018-11-13 20 51 55

Compared with the other SWAT Suits
screenshot 2018-11-13 20 51 52

Functionally the armour is identical to the Captains variant sans its greyscale. 

Current its not found anywhere in game and is admin only. I could imagine it being some kind of base for a hirable Mercenary that admins could send in for events or otherwise just a misc item that might show up or be used later. I don't think it'd be appropriate for the station to be able to buy one.

:cl: Steelpoint
add: MK II SWAT Suit has been officially added to the game, the base model the Captains SWAT Suit is based off of. Currently it does not spawn in the game anywhere and is admin only. 
/:cl:

### CODE REVIEW

**Comment by Arianya** [comment]
Wouldn't it make more sense to make this suit the parent and then make the captain's one a child of it?

**Comment by Steelpoint** [comment]
Actually a good point. I'll rework it to do that when I get a chance.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm on the fence about that sprite, there's so much going on i'm not sure what's going on!

**Comment by SgtxRootbeer** [comment]
The legs and sides of the chest look poorly defined, and the helmet's eyepiece looks different from the Captain's.

Also, you might wanna add the ability to throw the mkII into a washing machine with a blue crayon to get a Captain's varient.

**Comment by GravitronBlue** [comment]
If this is of any relevance I made a somewhat polished Mk2 suit a while back after being unsatisfied with the greyscale icon.

(It's really just a less stylish Deathsquad suit that fits better with original UEG helmet icon)

**Comment by ExcessiveUseOfCobblestone** [comment]
having some color to help define the suit really helps, nice edits.

Reminder you can put sprites in our sprite repo for people to use, i'm sure coders would love to have them :3

**Comment by optimumtact** [comment]
I have an inbuilt rage defence mechanism against swat suits thanks to @Bawhoppen

**Comment by Bawhoppen** [comment]
SWAT is the future, don't let anyone tell you otherwise @Steelpoint

**Comment by Steelpoint** [comment]
@GravitronBlue if you want me to use your sprites I'd need em in a dmi format for ease of use.

**Comment by Steelpoint** [comment]
I've repathed it so the Captains SWAT suit is now a subtype of the regular SWAT suit.

**Comment by ExcessiveUseOfCobblestone** [comment]
why did you ping baw you've made a grave mistake...

**Comment by FantasticFwoosh** [comment]
> 
> 
> SWAT is the future, don't let anyone tell you otherwise @Steelpoint

You're either SWAT or you're not

**Comment by lolman360** [comment]
craftable/orderable secways when

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -528,36 +528,46 @@
...
 	armor = list("melee" = 40, "bullet" = 50, "laser" = 50, "energy" = 25, "bomb" = 50, "bio" = 100, "rad" = 50, "fire" = 100, "acid" = 100)
 	resistance_flags = FIRE_PROOF | ACID_PROOF
 	heat_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT //this needed to be added a long fucking time ago
-	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/captain
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/swat
 
-/obj/item/clothing/suit/space/hardsuit/captain/Initialize()
-	. = ..()
-	allowed = GLOB.security_hardsuit_allowed
```
Comment (SpaceManiac) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Seems like this got dropped

**Review by SpaceManiac — DISMISSED** [review]

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 1 above)
Comment (Steelpoint) [reply code/modules/clothing/spacesuits/hardsuit.dm]: I think I resolved it

**Comment by Steelpoint** [comment]
I think I should've fixed the issue.

**Review by ExcessiveUseOfCobblestone — DISMISSED** [review]

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -528,37 +528,50 @@
...
+	//Captain
+/obj/item/clothing/head/helmet/space/hardsuit/swat/captain
+	name = "captain's hardsuit helmet"
+	icon_state = "capspace"
+	item_state = "capspacehelmet"
+	desc = "A tactical SWAT helmet MK.II boasting better protection and a horrible fashion sense."
+
+/obj/item/clothing/suit/space/hardsuit/swat/captain
+	name = "captain's SWAT suit"
+	desc = "A MK.II SWAT suit with streamlined joints and armor made out of superior materials, insulated against intense heat. The most advanced tactical armor available Usually reserved for heavy hitter corporate security, this one has a regal finish in Nanotrasen company colors. Better not let the assistants get a hold of it."
```
Comment (SpaceManiac) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Missing `.` before `Usually`

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 3]
```diff
@@ -528,37 +528,50 @@
...
-	return //Sprites required for flashlight
+/obj/item/clothing/head/helmet/space/hardsuit/swat/attack_self()
 
-/obj/item/clothing/suit/space/hardsuit/captain
-	name = "captain's SWAT suit"
-	desc = "A MK.II SWAT suit with streamlined joints and armor made out of superior materials, insulated against intense heat. The most advanced tactical armor available. Usually reserved for heavy hitter corporate security, this one has a regal finish in Nanotrasen company colors. Better not let the assistants get a hold of it."
-	icon_state = "caparmor"
-	item_state = "capspacesuit"
+/obj/item/clothing/suit/space/hardsuit/swat
+	name = "MK II. SWAT Suit"
```
Comment (SpaceManiac) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Name should have `\improper`, so we get "a/the MK.II SWAT suit"

**Review by SpaceManiac — DISMISSED** [review]
Is it "MK II." or "MK.II"? It's meant to be in the same series as the "MK.I" and "MK.III", right?

**Comment by Steelpoint** [comment]
done

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 4]
```diff
@@ -528,37 +528,50 @@
...
 
-	//Captain
-/obj/item/clothing/head/helmet/space/hardsuit/captain
-	name = "captain's hardsuit helmet"
-	icon_state = "capspace"
-	item_state = "capspacehelmet"
-	desc = "A tactical SWAT helmet MK.II boasting better protection and a horrible fashion sense."
+	//SWAT MKII
+/obj/item/clothing/head/helmet/space/hardsuit/swat
+	name = "MK.II. SWAT Helmet"
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/hardsuit.dm]:
```suggestion
	name = "\improper MK.II. SWAT Helmet"
```

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 5]
```diff
@@ -528,37 +528,50 @@
...
-	return //Sprites required for flashlight
+/obj/item/clothing/head/helmet/space/hardsuit/swat/attack_self()
 
-/obj/item/clothing/suit/space/hardsuit/captain
-	name = "captain's SWAT suit"
-	desc = "A MK.II SWAT suit with streamlined joints and armor made out of superior materials, insulated against intense heat. The most advanced tactical armor available. Usually reserved for heavy hitter corporate security, this one has a regal finish in Nanotrasen company colors. Better not let the assistants get a hold of it."
-	icon_state = "caparmor"
-	item_state = "capspacesuit"
+/obj/item/clothing/suit/space/hardsuit/swat
+	name = "MK.II. SWAT Suit"
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/hardsuit.dm]:
```suggestion
	name = "\improper MK.II. SWAT Suit"
```

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 6]
```diff
@@ -528,37 +528,50 @@
...
+/obj/item/clothing/suit/space/hardsuit/swat/Initialize()
 	. = ..()
 	allowed = GLOB.security_hardsuit_allowed
 
+	//Captain
+/obj/item/clothing/head/helmet/space/hardsuit/swat/captain
+	name = "captain's hardsuit helmet"
+	icon_state = "capspace"
+	item_state = "capspacehelmet"
+	desc = "A tactical SWAT helmet MK.II boasting better protection and a horrible fashion sense."
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/hardsuit.dm]:
```suggestion
	desc = "A tactical MK.II SWAT helmet boasting better protection and a horrible fashion sense."
```

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 7]
```diff
@@ -528,37 +528,50 @@
...
+
+/obj/item/clothing/suit/space/hardsuit/swat/captain
+	name = "captain's SWAT suit"
+	desc = "A MK.II SWAT suit with streamlined joints and armor made out of superior materials, insulated against intense heat. The most advanced tactical armor available Usually reserved for heavy hitter corporate security, this one has a regal finish in Nanotrasen company colors. Better not let the assistants get a hold of it."
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/hardsuit.dm]:
```suggestion
	desc = "A MK.II SWAT suit with streamlined joints and armor made out of superior materials, insulated against intense heat. The most advanced tactical armor available. Usually reserved for heavy hitter corporate security, this one has a regal finish in Nanotrasen company colors. Better not let the assistants get a hold of it."
```

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
finishing touches :)

**Comment by Steelpoint** [comment]
Yay

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by Steelpoint** [comment]
The offer to use @GravitronBlue sprites in lieu of this PR's one is still on the table.

**Pull request merged**